dask_ndfourier/core.py

```diff
@@ -1 +1,109 @@
 # -*- coding: utf-8 -*-
+
+
+import collections
+import numbers
+
+import numpy
+
+import dask.array
+
+from dask_ndfourier import _compat
+
+try:
+    from itertools import imap
+except ImportError:
+    imap = map
+
+
+def fourier_shift(input, shift, n=-1, axis=-1):
+    """
+    Multi-dimensional fourier shift filter.
+
+    The array is multiplied with the fourier transform of a shift operation.
+
+    Parameters
+    ----------
+    input : array_like
+        The input array.
+    shift : float or sequence
+        The size of the box used for filtering.
+        If a float, `shift` is the same for all axes. If a sequence, `shift`
+        has to contain one value for each axis.
+    n : int, optional
+        If `n` is negative (default), then the input is assumed to be the
+        result of a complex fft.
+        If `n` is larger than or equal to zero, the input is assumed to be the
+        result of a real fft, and `n` gives the length of the array before
+        transformation along the real transform direction.
+    axis : int, optional
+        The axis of the real transform.
+
+    Returns
+    -------
+    fourier_shift : Dask Array
+
+    Examples
+    --------
+    >>> from scipy import ndimage, misc
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy.fft
+    >>> fig, (ax1, ax2) = plt.subplots(1, 2)
+    >>> plt.gray()  # show the filtered result in grayscale
+    >>> ascent = misc.ascent()
+    >>> input_ = numpy.fft.fft2(ascent)
+    >>> result = ndimage.fourier_shift(input_, shift=200)
+    >>> result = numpy.fft.ifft2(result)
+    >>> ax1.imshow(ascent)
+    >>> ax2.imshow(result.real)  # the imaginary part is an artifact
+    >>> plt.show()
+    """
+
+    if issubclass(input.dtype.type, numbers.Real):
+        input = input.astype(complex)
+
+    # Validate and normalize shift
+    if isinstance(shift, numbers.Number):
+        shift = input.ndim * [shift]
+    elif not isinstance(shift, collections.Sequence):
+        raise TypeError("The `shift` must be a number or a sequence.")
+    if len(shift) != input.ndim:
+        raise RuntimeError(
+            "The `shift` must have a length equal to the input's rank."
+        )
+    if all(imap(lambda i: isinstance(i, numbers.Integral), shift)):
+        pass
+    elif all(imap(lambda i: isinstance(i, numbers.Real), shift)):
+        raise NotImplementedError("Real value(s) unsupported in the `shift`.")
```

Comment:
Opened issue ( https://github.com/dask-image/dask-ndfourier/issues/4 ) to add support for this down the road.